tests/test_queuing.py

```diff
@@ -285,37 +289,145 @@ def template(x):
         assert str(recorder) == expected_output
 
 
-class AnnotatingTensor(qml.operation.Tensor):
-    """Dummy tensor class that queues itself on initialization
-    to an annotating queue."""
+class TestAnnotatedQueue:
+    """Tests for the annotated queue class"""
 
-    def __init__(self, *args):
-        super().__init__(*args)
-        self.queue()
+    def test_remove_not_in_queue(self):
+        """Test that remove does not fail when the object to be removed is not in the queue."""
 
-    def queue(self):
-        qml.QueuingContext.append(self)
+        with qml._queuing.AnnotatedQueue() as q1:
+            op1 = qml.PauliZ(0)
+            op2 = qml.PauliZ(1)
+            q1.append(op1)
+            q1.append(op2)
```

Comment:
Since these tests are all for an `AnnotatedQueue`, and are duplicated in the parent class, might be worthwhile to pass some junk kwargs as well to make them test something different?